tachyon_core/src/storage/writer/persistent_writer.rs

```diff
@@ -29,7 +30,7 @@ impl PersistentWriter {
         ts: Timestamp,
         v: Value,
         value_type: ValueType,
-    ) -> PartiallyPersistentDataFile {
+    ) -> Result<PartiallyPersistentDataFile, WriterErr> {
         let open_file = self
```

Comment:
Curious why the API allows multiple open files `get_open_files_for_stream_id` to be fetched? Should we assert that length is 1?

Reply:
I previously had an assert statement there but removed it in a previous PR, I put it back. In a different PR we should make a list of "Fatal" errors that we can't recover from instead of using things like asserts. This could also include the panic for magic number